### DISCUSSION TITLE
ColumnIterator in reverse order

### DISCUSSION THREAD

**Discussion by wolverine4277** [PHPOffice/PhpSpreadsheet#4100] [General]
Hi, I'm trying to use iterators to go through columns or rows in reverse order but I can't find a way to do it.
So far I have tried the following (I'm using a simple xls with columns from A to N):

```
$spreadsheet = \PhpOffice\PhpSpreadsheet\IOFactory::load($sourceXlsx);
$sheet = $spreadsheet->getActiveSheet();
$highestColumn = $sheet->getHighestColumn();
$columnIterator = $sheet->getColumnIterator($highestColumn, 'A');
foreach ($columnIterator as $column) {
    echo $column->getColumnIndex();
}
```
The above code only print the last column N.
If i use something like this:

```
$spreadsheet = \PhpOffice\PhpSpreadsheet\IOFactory::load($sourceXlsx);
$sheet = $spreadsheet->getActiveSheet();
$highestColumn = $sheet->getHighestColumn();
$columnIterator = new \ArrayIterator(range($highestColumn, 'A'));
$columnIterator = new \LimitIterator($columnIterator, 0, count($columnIterator));
foreach ($columnIterator as $column) {
    echo $column;
}

```
The above code prints correctly NMLKJIHGFEDCBA but if I use a xls with many columns the code throws my this error:

> Warning: range(): Argument #1 ($start) must be a single byte, subsequent bytes are ignored

in line:

`new \ArrayIterator(range($highestColumn, 'A'));`

Evidently something I'm not taking into account but I can't figure it out.
I would appreciate any help.
Thanks!

**Comment by oleibman** [comment]
Can you make use of the following answer from [@mpen](https://stackoverflow.com/users/65387/mpen) in https://stackoverflow.com/questions/5315539/iterate-in-reverse-through-an-array-with-php-spl-solution:
```php
/**
 * Iterate an array or other foreach-able without making a copy of it.
 *
 * @param array|\Traversable $iterable
 * @return Generator
 */
function iter_reverse($iterable) {
    for (end($iterable); ($key=key($iterable))!==null; prev($iterable)){
        yield $key => current($iterable);
    }
}
//Usage:
foreach(iter_reverse($my_array) as $key => $value) {
    // ... do things ...
}
```
